### PR TITLE
Refactor/144 cluster config

### PR DESCRIPTION
**Pull request by philwinder**
A major refactor to improve the use of the configuration object. See #144 and documentation in com.containersol.minimesos.mesos.ClusterArchitecture.

Old classes have been depreciated, but retained so that old code _should_ still work. (Will test this now).

### CODE REVIEW

File: src/main/java/com/containersol/minimesos/MesosCluster.java [hunk 1]
```diff
@@ -52,50 +45,55 @@
...
+    private ClusterArchitecture clusterArchitecture;
 
-    protected String zkUrl;
-
-    protected ZooKeeper zkContainer;
-
-    private String clusterId;
+    private static String clusterId;
 
+    @Deprecated
```
Comment (mwl) [inline src/main/java/com/containersol/minimesos/MesosCluster.java]: A little comment on where to looks for it's replacement?

File: build.gradle [hunk 2]
```diff
@@ -33,6 +33,7 @@ tasks.withType(JavaCompile) {
 }
 
 repositories {
+    jcenter()
```
Comment (mwl) [inline build.gradle]: Why jcenter? mockito is in maven central too.

**Comment by remmelt** [comment]
If we can get rid of the `@Deprecated` annotations and the methods/classes they refer to, I'd vote to do that. Unless there are people depending on these features and we want to ease them into the new way, let's just get on with it.

**Comment by jrslv** [comment]
:+1:

**Comment by sadovnikov** [comment]
just to clarification, is anyone removing deprecated classes? updating text on http://minimesos.org/ and http://minimesos.readthedocs.org/en/latest/ ?

**Pull request merged**